src/util/audio/VorbisProducer.cpp

```diff
@@ -32,21 +33,27 @@ bool VorbisProducer::start(std::string filename, unsigned int timestamp)
 	this->audioQueue->setAudioAttributes(this->sfInfo.samplerate, static_cast<unsigned int>(this->sfInfo.channels));
 
 	this->producerThread = new std::thread(
-			[&, filename]
+			[&, filename, seekPosition]
 			{
 				size_t numFrames = 1;
 				auto sampleBuffer = new float[1024 * this->sfInfo.channels];
 				std::unique_lock<std::mutex> lock(audioQueue->syncMutex());
-
+								
 				while (!this->stopProducer && numFrames > 0 && !this->audioQueue->hasStreamEnded())
 				{
-			        numFrames = sf_readf_float(this->sfFile, sampleBuffer, 1024);
+					numFrames = sf_readf_float(this->sfFile, sampleBuffer, 1024);
 
 					while (this->audioQueue->size() >= this->sample_buffer_size && !this->audioQueue->hasStreamEnded() && !this->stopProducer)
 					{
 						audioQueue->waitForConsumer(lock);
 					}
 
+					if (this->seekSeconds != 0)
+					{
+						sf_seek(this->sfFile,seekSeconds * this->sfInfo.samplerate, SEEK_CUR);
+						this->seekSeconds = 0;
```

Comment:
This creates a race condition on the variable `seekSeconds` but I think we can live with it as users are unable to click nearly as fast as this poll and reset happens.

Reply:
Agreed. The worst that can happen is the program skipping a seek command if the user can click twice in a few nanoseconds (which I'm not even sure it's possibile to catch on a non real-time O.S.) so I wouldn't complicate it with synchronization constructs